### ISSUE TITLE
std::optional vs boost::optional

### ISSUE THREAD

**Issue by sjunges** [moves-rwth/storm#302]
did we agree which we use? I am slightly in favor of std.

**Comment by tquatmann** [comment]
My vote is for `std::optional` as well.

**Comment by volkm** [comment]
I am also in favour of using features from std whenever possible.

**Comment by volkm** [comment]
As a follow-up, other constructs such as `boost::any` can possibly also be replaced.

**Comment by tquatmann** [comment]
Agreed; I'm not sure, however, if we should should replace all the occurrences at once or do this slowly, whenever we work on a file? I prefer the latter, tbh.

**Comment by volkm** [comment]
I personally would prefer the former. Then we have one dedicated commit which changes this behaviour, and we can also make sure that certain constructs are not occurring anymore. Otherwise, I could see that certain parts are never updated, because the file is never worked on.
I see however, that the first option is more work at once and one has to be more careful not to break anything.

**Comment by sjunges** [comment]
Having done this a bit: one dedicated commit is definitively chaos. I think it is good to start getting particular parts of the code boost-optional free, that way, it feels rewarding, but also something that you can do in a few unproductive hours.

**Comment by volkm** [comment]
Some related discussion [here](https://github.com/moves-rwth/storm/pull/531)
